### PR TITLE
Custom URL tweaks

### LINKED ISSUE #1368: Add migration for site list metadata, converting website names to url_paths
Related to #1336, course lists need to save the `url_path` instead of the `name` to metadata.  A migration or management command is needed to update existing course lists for ocw-www.  The same hypothetically should be done for resource lists, but none exist yet on production.

### LINKED ISSUE #1369: Resource collections need to save website url_path instead of name to metadata
Related to #1336, resource collections need to save `Website.url_path` instead of `Website.name` to metadata (and the corresponding hugo themes template needs to change as well).  This requires a changed to the `RelationField` widget.

### LINKED ISSUE #1370: Custom URL tweaks
- The publishing success/failure email message contains the wrong link for ocw-www and needs to be fixed
- The video transcript download links are wrong and need to be changed.
- When publishing to draft a subsequent time, there should be a clickable link above the URL field (assuming it hasn't been published to production yet, otherwise there won't be a URL field, just a link).

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #1370 
Closes #1369 
Closes #1368 

#### What's this PR do?
- Saves `Website.url_path` instead of `Website.name` to resource collections and course lists, migrates data for existing ones.
- Modifies the video transcript metadata to point to the correct url based on parent website url
- Includes a working link above the URL field when publishing a subsequent time, but before a production publish.
- Fixes the link in ocw-www publishing success/failure emails

#### How should this be manually tested?
- Create some resource collections and course-lists for your www site, include sites where the name and url_path differ, reopen them after editing, make sure they work as expected, and that the values saved to git are for the `url_path` and not `name`.  Also check that any pre-existing resource collections and course-lists have been updated to use `url_path` (the git metadata/markdown should show `courses/<path_url>` instead of just `<name>`)
- Follow test instructions for https://github.com/mitodl/ocw-studio/pull/1345, but at the end publish a couple times with different url paths and ensure the caption/transcript path metadata gets updated with the new path each time.
- Create a new video gallery for a site and ensure it still works.
- Create a new site, choose a url, publish to draft (it will fail, that's ok), then open the publish drawer again.  This time there should be a real link above the URL field.
- Test out with [this branch of ocw-hugo-themes](https://github.com/mitodl/ocw-hugo-themes/pull/700) to ensure everything renders properly with the new values for resource collections, video galleries, course lists, search, etc.

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
Screen Shot 2022-05-19 at 1 11 05 PM

#### What GIF best describes this PR or how it makes you feel?
(Optional)

### CODE REVIEW

File: websites/serializers.py [hunk 1]
```diff
@@ -548,6 +559,7 @@ def get_content_context(self, instance):  # pylint:disable=too-many-branches
                     website__name=website_name, text_id__in=text_ids
```
Comment (ChristopherChudzicki) [inline websites/serializers.py:559]:
```py
        for website_url_path, text_ids in lookup.items():
            contents.extend(
                WebsiteContent.objects.filter(
                    website__url_path=website_url_path, text_id__in=text_ids
                )
            )
```

This made cross-site relation widget work, but broke the same-site version that is used in, e.g., video galleries.
Screen Shot 2022-05-20 at 11 28 35 AM

File: videos/tasks.py [hunk 2]
```diff
@@ -216,7 +216,9 @@ def delete_s3_objects(
 def update_transcripts_for_video(video_id: int):
     """Update transcripts for a video"""
     video = Video.objects.get(id=video_id)
+    log.error("update_transcripts_for_video")
```
Comment (gumaerc) [inline videos/tasks.py:219]: Seems like we maybe have some stray debug logging here?

**Review by gumaerc — CHANGES_REQUESTED** [review]
I was able to fully test this and the `name` and `id` properties have been changed to `url_path` where expected.  I tested by pulling a full backup of the production `ocw-studio` database and overwriting my local copy, adding a test site on top of that where I customized the URL.  After making some adjustments to https://github.com/mitodl/ocw-hugo-themes/pull/700 I was able to get `ocw-www` to render properly.  It's difficult to fully test rendering the site locally.  When you publish to a test org and clone that and `ocw-www` to build locally, your local Concourse has nowhere to deploy the site.  This means what when you actually try to view the course in any lists that it is added to, you will encounter a build error because of trying to query static JSON for a course that doesn't exist according to either RC or Production.

Anyway that is a separate issue with local development and testing that I'm hoping we can smooth out.  Something I want to do soon is refactor our use of `getJSON` to `resources.GetRemote` so we can utilize error handling when fetching remote resources.  As far as this code review is concerned, I only noticed what I think are some stray debug logging lines.  Other than that, we should get this into RC so we can properly test the themes PR and then get that out ASAP.

File: videos/tasks.py (hunk 2 above)
Comment (mbertrand) [reply videos/tasks.py]: removed

**Pull request merged**